docs/content/docker.md

```diff
@@ -0,0 +1,280 @@
+
+This is great for creating standalone volumes and easily connecting them to
+directories in containers as a way to share data between multiple containers.
+It also allows for the use of Docker Volume Plugins with stateful containers
+across multiple hosts. It can be useful to have persistent data on your host.
+
+Unlike the default `local` driver that comes with Docker,
+your data *will not be deleted* when the Volume is removed.
+With the `rclone` driver, if you remove the driver, and then recreate it
+later, any volume that was added to that volume will *still be there*.
+
+According to the [Docker Plugin API Docs](https://docs.docker.com/engine/extend/plugin_api/)
+plugins can be run managed by the docker daemon or as a native system service.
+The former mode is preferred, so we describe it first.
+
+## Running as Managed Plugin
+
+Docker daemon can install plugins from an image registry and run them managed.
+We maintain the
+[docker-volume-rclone](https://hub.docker.com/r/rclone/docker-volume-rclone/)
+plugin image on the [Docker hub](https://hub.docker.com).
+
+Before installing the plugin, let's create a couple required directories as
+plugin will *not* create them automatically:
+```
+mkdir -p /var/lib/docker/plugins/rclone/config
+mkdir -p /var/lib/docker/plugins/rclone/cache
+```
+
+Now install and enable the plugin:
+```
+docker plugin remove rclone
+docker plugin install rclone/docker-volume-rclone --grant-all-permissions --disable --alias rclone
+docker plugin set rclone RCLONE_VERBOSE=2 config=/etc/rclone args="--vfs-cache-mode=writes --allow-other"
+docker plugin enable rclone
+```
+
+The `-v /run/docker/plugins/:/run/docker/plugins/` part will make sure the
+`sock` file gets created at the right place. You also need to add one or more
+volumes to places you want to mount your volumes later at.
+
+**Note:** Plugin will fail if you make args empty, use e.g. "args=-v",
+due to the [docker bug](github/moby/plugin/v2/plugin.go:195).
+
+Lastly, the `-v /path/to/store/json/for/restart/:/var/lib/docker/plugin-data/`
+part is so that the plugin can create a `json` file to know what volumes
+existed in case of a system restart, etc.
+
+When the container is destroyed, etc, it will look at a file it created in
+`/var/lib/docker/plugins/rclone/docker-plugin.state` to recreate any volumes
+that had previously existed, so you want that JSON file to persist on the host.
+
+## Creating Volumes
+
+You can create a volume with this plugin:
+```
+docker volume list
+docker volume create vol1 -d rclone -o remote=storj: -o vfs-cache-mode=full
+docker volume create vol2 -d rclone -o remote=:tardigrade,access_grant=xxx:heimdall
+docker volume create vol3 -d rclone -o type=tardigrade -o path=heimdall -o tardigrade-access-grant=xxx -o poll-interval=0
+docker volume inspect vol1
+```
+
+When you create a container, you can connect it to the Volume:
+```
+docker run --rm -it -v vol1:/mnt --workdir /mnt ubuntu:bionic
+```
```

Comment:
I should mention that after the containter using rclone volume is destroyed and/or recreated by swarm manager on another worker host, the volume will be unmounted but the volume definition will not be removed by docker from the host. If you absolutely want to get rid of the volume, you must `docker volume rm myvol` it manually.